chiselFrontend/src/main/scala/chisel3/Printable.scala

```diff
@@ -157,22 +161,40 @@ object FirrtlFormat {
   }
 }
 /** Format bits as Decimal */
-case class Decimal(bits: Bits) extends FirrtlFormat('d')
+case class Decimal(bits: Bits) extends FirrtlFormat('d') {
+  override def length: Int = bits.widthOption match {
+    case Some(w) => Math.ceil(Math.log10(Math.pow(2, w) + 1)).toInt
+    case None => 0
```

Comment:
Should the `None` case be an error, like it is for `FullName`?  Or alternatively should `length`'s type be `Option[Int]`?

Reply:
Perhaps. The reason I made it zero was because it seems like an estimate of the length (e.g. all the other widths in an aggregate) would potentially be preferable to a None. I don't have a strong preference, though.

Reply:
I see.  If it's only an estimate, I don't have any objections, but it might be worth indicating that fact in the scaladoc for `length`.